src/chart_process/bms_processor.rs

```diff
@@ -61,7 +65,11 @@ impl BmsProcessor {
         let default_visible_y_length =
             compute_default_visible_y_length(base_bpm.clone(), reaction_time);
 
-        let all_events = Self::precompute_all_events::<T>(&bms);
+        let pre_index = {
+            let bms: &Bms = &bms;
+            BmsPrecomputer::precompute_all_events::<T>(bms)
+        };
```

Comment:
Why don't you simply do this?
```suggestion
        let pre_index = BmsPrecomputer::precompute_all_events::<T>(&bms);
```